src/igvfd/types/file_set.py

```diff
@@ -190,6 +191,34 @@ def submitted_files_timestamp(self, request, files, auxiliary_sets=[]):
     def input_for(self, request, input_for):
         return paths_filtered_by_status(request, input_for)
 
+    @calculated_property(
+        condition='samples',
+        schema={
+            'title': 'Construct Library Sets',
+            'description': 'The construct library sets associated with the samples of this file set.',
+            'type': 'array',
+            'minItems': 1,
+            'uniqueItems': True,
+            'items': {
+                'title': 'Construct Library Set',
+                'description': 'A construct library set associated with a sample of this file set.',
+                'type': 'string',
+                'linkTo': 'FileSet',
+            },
+            'notSubmittable': True
+        })
+    def construct_library_sets(self, request, samples=None):
+        construct_library_sets = set()
+        for sample in samples:
+            sample_object = request.embed(sample,
+                                          '@@object_with_select_calculated_properties?'
+                                          'field=construct_library_sets'
+                                          )
+            if sample_object.get('construct_library_sets', []):
+                construct_library_sets = construct_library_sets | set(sample_object.get('construct_library_sets', []))
```

Comment:
are you only keeping the CSL for the first sample in samples?

Reply:
construct_library_sets = construct_library_sets | set(sample_object.get('construct_library_sets', [])) adds to the existing set

Reply:
oh thanks, forgot it's the union function for set